File: 0001_two_sum/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  const map = new Map();
  nums.forEach((e, i) => map.set(e, i));
  
  for (let i in nums) {
    let j = map.get(target - nums[i]);
    if (j !== undefined && j != i) {
      return [i, j];
    }
  }
};

File: 0001_two_sum/two_sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val indexed = nums.zip(nums.indices)
        val map = indexed.toMap()

        return indexed
            .mapNotNull { (v, i) ->
                map[target - v]
                    ?.takeIf { it != i } 
                    ?.let { intArrayOf(i, it) }
            }
            .first()
    }
}
